### PR TITLE
refactor(flutter_login): dispose authentication repository

### PR DESCRIPTION
**Pull request by bensonarafat**
Repository dispose updated 

## Breaking Changes

NO

## Description
AuthenticationRepository shouldn't be disposed from bloc. 
So best practice [comment](https://github.com/felangel/bloc/issues/952#issuecomment-598793189) by @felangel  

create a StatefulWidget to create the repository and handle releasing resources in the dispose.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: examples/flutter_login/lib/app.dart [hunk 1]
```diff
@@ -17,14 +17,25 @@ class App extends StatelessWidget {
...
   final UserRepository userRepository;
 
+  @override
+  State<App> createState() => _AppState();
+}
+
+class _AppState extends State<App> {
+  @override
+  void dispose() {
+    widget.authenticationRepository.dispose();
```
Comment (felangel) [inline examples/flutter_login/lib/app.dart:27]: Thanks for the pull request! Ideally I think the rule of thumb should be whoever creates the object should be responsible for disposing the object. In this case, the App does not create the authentication repository instance so it should not be responsible for disposing it. Can we move the disposable up to where the repository was created? Thanks!

Comment (bensonarafat) [reply examples/flutter_login/lib/app.dart:27]: Oh! Sure 
Can move it.. thanks

**Comment by bensonarafat** [comment]
The disposal moved upward. 
you can now review 
thanks

**Comment by bensonarafat** [comment]
@felangel Is there something missing that I should change in this PR?

**Comment by felangel** [comment]
> @felangel Is there something missing that I should change in this PR?

sorry for the delay, I'll get this merged later today 👍

**Comment by bensonarafat** [comment]
@felangel pls can this be merge 🙏

**Review by felangel — APPROVED** [review]

**Pull request merged**